AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSCoreDataClient.m

```diff
@@ -0,0 +1,235 @@
+                 deviceTimeToLive:deviceTimeToLive
+                        operation:operation];
+        completionHandler(deletedDocument);
+      }
+
+      // Create/Replace operation.
+      else if ([kMSPendingOperationCreate isEqualToString:(NSString *)operation] ||
+               [kMSPendingOperationReplace isEqualToString:(NSString *)operation]) {
+
+        // Serialize incoming document into a JSON string.
+        NSDictionary *dictionary = [document serializeToDictionary];
+        NSString *jsonDocument;
+        NSError *jsonError;
+        NSData *jsonData = [NSJSONSerialization dataWithJSONObject:dictionary options:0 error:&jsonError];
+        if (!error) {
+          jsonDocument = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+        } else {
+          NSString *message = @"Error serializing document for local storage";
+          MSLogError([MSDataStore logTag], message);
+          completionHandler([[MSDocumentWrapper alloc] initWithDataStoreErrorCode:MSACDataStoreLocalStoreError
+                                                                     errorMessage:message
+                                                                       documentId:documentId]);
+          return;
+        }
+
+        // Create a deleted document record.
+        MSDocumentWrapper *createdOrUpdatedDocument = [[MSDocumentWrapper alloc] initWithDeserializedValue:document
+                                                                                                 jsonValue:jsonDocument
+                                                                                                 partition:token.partition
+                                                                                                documentId:documentId
+                                                                                                      eTag:cachedDocument.eTag
+                                                                                           lastUpdatedDate:cachedDocument.lastUpdatedDate
+                                                                                          pendingOperation:operation
+                                                                                                     error:nil];
+
+        // Update local store and return document.
+        [self updateLocalStore:token
+            currentCachedDocument:cachedDocument
+                newCachedDocument:createdOrUpdatedDocument
+                 deviceTimeToLive:deviceTimeToLive
+                        operation:operation];
+        completionHandler(createdOrUpdatedDocument);
+      }
+    }
+  });
+}
+
+/**
+ * Validate an operation.
+ *
+ *@param operation The operation.
+ *
+ *@return YES if the operation is valid; NO otherwise.
+ */
++ (BOOL)isValidOperation:(NSString *)operation {
+  return operation == nil || [kMSPendingOperationCreate isEqualToString:operation] ||
```

Comment:
Just return `operation` instead of `operation == nil` (I think that should automatically be cast to a `BOOL` but I might be wrong)